neqo-transport/src/packet/mod.rs

```diff
@@ -1357,6 +1367,14 @@ mod tests {
         assert_eq!(&vn, &SAMPLE_VN);
     }
 
+    #[test]
+    fn vn_do_not_repeat_client_grease() {
+        fixture_init();
+        let vn =
+            PacketBuilder::version_negotiation(SERVER_CID, CLIENT_CID, 0x0a0a0a0a, &Version::all());
+        assert_ne!(&vn[SAMPLE_VN.len() - 4..], &[0x0a, 0x0a, 0x0a, 0x0a]);
```

Comment:
Would it make sense to use something like [mockall](https://docs.rs/mockall/latest/mockall/) to mock `random`?

Reply:
We can try that. Can you make a patch?

Reply:
:+1: I'll give it a shot. My current thought is to try to [use automock on the module](https://docs.rs/mockall/latest/mockall/#modules), but it could require some amount of reorganization for the `neqo-crypto` interface for `random`.

Reply:
Took some time to look at automock and at this time I don't think they'd fit our needs here. Perhaps something to look into in the future though.